src/Units/KaijuActions.java

```diff
@@ -0,0 +1,943 @@
+        || INERT_CRUSHABLES.contains(location.getEnvironment().terrainType);
+  }
+
+  public static class KaijuCrushAction extends GameAction
+  {
+    private final KaijuCrushFactory type;
+    private final GamePath movePath;
+    private final XYCoord waitLoc;
+    private final Unit actor;
+
+    public KaijuCrushAction(Unit unit, GamePath path, KaijuCrushFactory pType)
+    {
+      type = pType;
+      actor = unit;
+      movePath = path;
+      if( (null != path) && (path.getPathLength() > 0) )
+      {
+        // Store the destination for later.
+        waitLoc = movePath.getEndCoord();
+      }
+      else
+        waitLoc = null;
+    }
+
+    @Override
+    public GameEventQueue getEvents(MapMaster gameMap)
+    {
+      // CRUSH actions consist of
+      //   sequence:
+      //     [Terrain destruction]
+      //      - [Heal for Big Donk]
+      //     [Mass damage]
+      //      - [Transform for Hell Turkey]
+      //     [Death]
+      //   MOVE
+      GameEventQueue crushEvents = new GameEventQueue();
+
+      // Validate input.
+      boolean isValid = true;
+      isValid &= null != actor && !actor.isTurnOver;
+      isValid &= (null != movePath) && (movePath.getPathLength() > 0);
+      isValid &= (null != gameMap);
+      isValid &= (actor.model instanceof KaijuUnitModel);
+
+      // Generate events.
+      if( isValid )
+      {
+        Utils.enqueueMoveEvent(gameMap, actor, movePath, crushEvents);
+        // Remove and discard the move event; we'll be making iterative ones
+        crushEvents.clear();
+
+        // Tracks the Kaiju's predicted state as it progresses through the crush events
+        UnitContext kaijuState = new UnitContext(gameMap, actor);
+        final int startingMP = kaijuState.calculateMovePower();
+
+        // movePath should be updated by the above, so we should be good to go
+        for( PathNode node : movePath.getWaypoints() )
+        {
+          if( kaijuState.movePower < 1 )
+            break; // Go only until we run out of move
+
+          enqueueSingleTileEvents(gameMap, startingMP, kaijuState, node, crushEvents);
+        } // ~node loop
+      }
+      return crushEvents;
+    }
+    public static void enqueueSingleTileEvents(MapMaster gameMap, int startingMP, UnitContext kaijuState, PathNode node, GameEventQueue crushEvents)
+    {
+      KaijuUnitModel stomperType = (KaijuUnitModel) kaijuState.model;
+      final XYCoord destCoord = node.GetCoordinates();
+      MapLocation location = gameMap.getLocation(destCoord);
+
+      Unit victim = location.getResident();
+      if( null != victim && kaijuState.unit != victim )
+      {
+        KaijuWarsUnitModel victimType = (KaijuWarsUnitModel) victim.model;
+        int counter = 0;
+        int stompDamage = victim.getHP();
+        if( victimType.isKaiju )
+        {
+          // When stomping another kaiju, at least one of us dies
+          counter += victim.getHP();
+          stompDamage = kaijuState.getHP();
+        }
```

Comment:
So a 10 HP Kaiju can kill a 1HP Kaiju for 1 damage, and a 1 HP Kaiju can murder/suicide a 10 HP Kaiju?

Reply:
Nope.
10 HP can run over 1 HP kaiju for one damage.
1 HP kaiju can kill itself to deal 1 damage to a 10 HP kaiju.